### PR TITLE
PP-2974 Direct Debit confirm

### PR DESCRIPTION
**Pull request by rauligs**
## WHAT

- Add relevant states and events for Sandbox confirm.
- Add Mandates database table as discussed in db design using at the moment external_id as reference.
- Add new ConflictException for when states transitions are not expected / invalid given the current state and the event.
- Pending next PR return mandate reference on confirm

### CODE REVIEW

File: src/main/resources/migrations/00006_create_table_mandates.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
+--liquibase formatted sql
+
+--changeset uk.gov.pay:add_table-mandates
+CREATE TABLE mandates (
+    id BIGSERIAL PRIMARY KEY,
+    payer_id BIGSERIAL NOT NULL,
+    external_id VARCHAR(30) NOT NULL,
```
Comment (simad) [inline src/main/resources/migrations/00006_create_table_mandates.sql]: We might need an index on external_id, but not right now it seems, so 👍

File: src/main/java/uk/gov/pay/directdebit/payments/model/PaymentRequestEvent.java [hunk 2]
```diff
@@ -75,32 +106,17 @@ public void setEventDate(ZonedDateTime eventDate) {
         CHARGE_CREATED,
         TOKEN_EXCHANGED,
         DIRECT_DEBIT_DETAILS_RECEIVED,
-        PAYER_CREATED;
+        PAYER_CREATED,
```
Comment (simad) [inline src/main/java/uk/gov/pay/directdebit/payments/model/PaymentRequestEvent.java]: Don't we want to add `MANDATE` as a possible Event Type? That was originally the plan, I reckon.

File: src/main/resources/migrations/00006_create_table_mandates.sql (hunk 1 above)
Comment (rauligs) [reply src/main/resources/migrations/00006_create_table_mandates.sql]: Absolutely! I'll add it now

File: src/main/java/uk/gov/pay/directdebit/payments/model/PaymentRequestEvent.java (hunk 2 above)
Comment (rauligs) [reply src/main/java/uk/gov/pay/directdebit/payments/model/PaymentRequestEvent.java]: Mandate is an Event type...

Comment (simad) [reply src/main/java/uk/gov/pay/directdebit/payments/model/PaymentRequestEvent.java]: Not according to the gods of source code :godmode:

**Pull request merged**